index.bs

```diff
@@ -617,7 +617,7 @@ option. If <code><a for="underlying source">type</a></code> is set to <code>unde
 
 <emu-alg>
   1. If ! IsReadableStream(*this*) is *false*, throw a *TypeError* exception.
-  1. Let _reader_ be ? AcquireReadableStreamDefaultReader(*this*).
+  1. Let _reader_ be ? Construct(`<a idl>ReadableStreamDefaultReader</a>`, « *this* »).
```

Comment:
This is a bit of a problem...
* If we create a reader with `forAuthorCode = false`, then `ReadableStreamCreateReadResult` inside `it.next()` will create an object with `null` prototype, which breaks the contract.
* If we create a reader with `forAuthorCode = false`, then `ReadableStreamDefaultReaderRead` inside `it.next()` will also access `Object.prototype.then` for the internal read result.

I think I'll have to rework this a bit. `ReadableStreamCreateReadResult` should take a `forAuthorCode` argument, rather than a `reader`. In most cases, we'll just pass `reader.[[forAuthorCode]]` as the argument, but in the case of `it.next()` we should explicitly pass `true`, even though `it.[[asyncIteratorReader]].[[forAuthorCode]]` will be `false`.